### ISSUE TITLE
Support forward proxies with CONNECT

### ISSUE THREAD

**Issue by edfungus** [grpc/grpc-java#2193]
Is there a way to specify a proxy connection when specifying the host and port with the NettyChannelBuilder? 

``` java
originChannel = NettyChannelBuilder.forAddress(host, port)
                          .sslContext(GrpcSslContexts.forClient().build())
                          .build();
```

**Comment by ejona86** [comment]
No, client-side proxies (a.k.a., forward proxies) are not currently supported. To understand your use-case, are you in an environment that requires using a client-side proxy to connect to the Internet? Or do you have a different reason to use a proxy?

**Comment by edfungus** [comment]
I see. 
Yes, because of the environment, we need to use a client-side proxy to connect to the internet. Would you happen to know a potential work around?

**Comment by ejona86** [comment]
I don't know of a reasonable workaround currently. It seems like you are using HTTPS. That is much easier to use the proxy for (because it would be CONNECT and the proxy doesn't need HTTP/2). Does your use case exclusively use HTTPS?

**Comment by edfungus** [comment]
Yes, our use case exclusively requires HTTPS

**Comment by edfungus** [comment]
Is proxies in the pipeline for a future release?

**Comment by ejona86** [comment]
This is being done for C in grpc/grpc#7611. There's consensus that it should be done for all the implementations. I don't have a timeline, but it is in the pipeline.

We aren't planning SOCKS or such, just CONNECT support.

**Comment by gvalva** [comment]
I am having the same issue. Does anyone know any workaround for j2ee applications.

**Comment by ejona86** [comment]
This is blocked by #4029. You can use GRPC_PROXY_EXP=host:port environment variable today. But the final solution is to use Java's ProxySelector. Today, it works for custom ProxySelectors that return resolved addresses. That issue will resolve selectors returning non-resolved addresses when using Netty.

**Comment by ejona86** [comment]
This was resolved in #4137. grpc observes the default Java proxy settings.
